parser.js

```diff
@@ -241,7 +241,8 @@ function parseGraph(graph, options)
 			// Set the context to the graph name
 			if (setNodeContext && parent)
 			{
-				node.parent = Utils.getIdFromResource(parent);
+				const parentId = Utils.getIdFromResource(parent);
+				node.parent = parentId !== node.id ? parentId : node.parent;
```

Comment:
This seems a like convoluted. Isn't it the case that `node.parent` is always `null` (given `node` was just created at line 237)?

Reply:
You are right! I haven't noticed that the node had just been created and therefore the `node.parent` will always be `null`. Just pushed a commit changing the default value to `null`.